www/js/callbacks.js

```diff
@@ -1019,6 +1019,54 @@ Callbacks = {
         CSEMOTELIST.handleChange();
     },
 
+    renameEmote: function (data) {
+        var badBefore = /\s/g.test(data.old);
+        var badAfter = /\s/g.test(data.name);
+        var oldName = data.old;
+        delete data.old;
+
+        data.regex = new RegExp(data.source, "gi");
+
+        for (var i = 0; i < CHANNEL.emotes.length; i++) {
+            if (CHANNEL.emotes[i].name === oldName) {
+                CHANNEL.emotes[i] = data;
```

Comment:
I'm sure you've realized by now that is indeed covered a bit lower into the function.